cosmwasm/contracts/v1/compute-tests/test-compute-contract/src/contract.rs

```diff
@@ -648,11 +645,44 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> S
         }
         ExecuteMsg::IncrementAndSendFailingSubmessage { reply_on } => {
             increment_simple(deps)?;
-            let mut response = send_failing_submsg(env, reply_on)?;
-            response.data = Some((count as u32).to_be_bytes().into());
+            let response = send_failing_submsg(env, reply_on)?;
 
             Ok(response)
         },
+        ExecuteMsg::IncrementAndSendSubmessageWithBankFail { reply_on } => {
+            increment_simple(deps)?;
+            let response = send_failing_submsg_with_bank_fail(env, reply_on)?;
+
+            Ok(response)
+        },
+        ExecuteMsg::SendSucceedingSubmessageThenFailingMessageOnReply {} => {
+            increment_simple(deps)?;
+
+            let mut response = Response::default();
+            add_succeeding_submsg(env, &mut response, ReplyOn::Always, 9201);
+
+            // failing message is created on reply
+            Ok(response)
+        },
+        ExecuteMsg::SendSucceedingSubmessageAndFailingMessage {} => {
+            increment_simple(deps)?;
+
+            let mut response = Response::default();
+
+            add_succeeding_submsg(env, &mut response, ReplyOn::Always, 9201);
+
+            response = response.add_message(
+                CosmosMsg::Bank(BankMsg::Send {
+                    to_address: "non-existent".to_string(),
+                    amount: vec![Coin::new(100, "non-existent")],
+                })
+            );
+
+            // NOTE that if the submsg is added after the message, then it is processed in THAT order,
+            // which is different from what is stated in cosmwasm's docs here: https://github.com/CosmWasm/cosmwasm/blob/main/SEMANTICS.md#order-and-rollback
```

Comment:
Interesting note here that I think describes a change in our behaviour compared to cosmwasm's docs!
@toml01
@liorbond